### ISSUE TITLE
False-positive error for `Final` if it's aliased

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#316]
flake8-pyi currently emits a false-positive error for this code, which should be fine:

```python
from typing_extensions import Final as _Final

CONSTANT: _Final = 5
```

This might be tricky to fix, since we don't currently track whether imports are aliased at all. Came up in https://github.com/python/typeshed/pull/9470

**Comment by Avasam** [comment]
Similarly, `Y036` (Badly defined __exit__ method) will false-positive with a TypeAlias of `object` (ie: `_Unused: TypeAlias = object`)

**Comment by AlexWaygood** [comment]
> Similarly, `Y036` (Badly defined __exit__ method) will false-positive with a TypeAlias of `object` (ie: `_Unused: TypeAlias = object`)

_Sure_, but I'm gonna say I _definitely don't_ want us to try and fix that one. It's ~reasonable to argue that we should track whether imports are aliased. But constructing a symbol table for the entire module would require whole new level of complexity 😃 it's important to remember that we're just a dumb linter, we're not a type checker 😁

**Comment by Avasam** [comment]
> it's important to remember that we're just a dumb linter, we're not a type checker

Right, you don't have easy access to that kind of information 😅
